editor/import/audio_stream_import_settings.cpp

```diff
@@ -0,0 +1,652 @@
+			if (current_beat > beat_max) {
+				beats_edit->set_value(beat_max);
+				stream->call("set_beat_count", beat_max);
+			}
+		}
+		stream->call("set_bar_beats", bar_beats_edit->get_value());
+	} else {
+		stream->call("set_bpm", 0);
+		stream->call("set_bar_beats", 4);
+		beats_enabled->hide();
+		beats_edit->hide();
+		bar_beats_label->hide();
+		bar_beats_edit->hide();
+	}
+	if (bpm_enabled->is_pressed() && beats_enabled->is_pressed()) {
+		stream->call("set_beat_count", beats_edit->get_value());
+	} else {
+		stream->call("set_beat_count", 0);
+	}
+
+	updating_settings = false;
+
+	_preview->update();
+	_indicator->update();
+	color_rect->update();
+}
+
+void AudioStreamImportSettings::_reimport() {
+	params["loop"] = loop->is_pressed();
+	params["loop_offset"] = loop_offset->get_value();
+	params["bpm"] = bpm_enabled->is_pressed() ? double(bpm_edit->get_value()) : double(0);
+	params["beat_count"] = (bpm_enabled->is_pressed() && beats_enabled->is_pressed()) ? int(beats_edit->get_value()) : int(0);
+	params["bar_beats"] = (bpm_enabled->is_pressed()) ? int(bar_beats_edit->get_value()) : int(4);
+
+	EditorFileSystem::get_singleton()->reimport_file_with_custom_parameters(path, importer, params);
+}
+
+AudioStreamImportSettings::AudioStreamImportSettings() {
+	get_ok_button()->set_text(TTR("Reimport"));
+	get_cancel_button()->set_text(TTR("Close"));
+
+	VBoxContainer *main_vbox = memnew(VBoxContainer);
+	add_child(main_vbox);
+
+	HBoxContainer *loop_hb = memnew(HBoxContainer);
+	loop_hb->add_theme_constant_override("separation", 4 * EDSCALE);
+	loop = memnew(CheckBox);
+	loop->set_text(TTR("Enable"));
+	loop->set_tooltip(TTR("Enable looping."));
+	loop->connect("toggled", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	loop_hb->add_child(loop);
+	loop_hb->add_spacer();
+	loop_hb->add_child(memnew(Label(TTR("Offset (in sec):"))));
+	loop_offset = memnew(SpinBox);
+	loop_offset->set_max(10000);
+	loop_offset->set_step(0.001);
+	loop_offset->set_tooltip(TTR("Loop offset (from beginning). Note that if BPM is set, this setting will be ignored."));
+	loop_offset->connect("value_changed", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	loop_hb->add_child(loop_offset);
+	main_vbox->add_margin_child(TTR("Loop:"), loop_hb);
+
+	HBoxContainer *interactive_hb = memnew(HBoxContainer);
+	interactive_hb->add_theme_constant_override("separation", 4 * EDSCALE);
+	bpm_enabled = memnew(CheckBox);
+	bpm_enabled->set_text((TTR("BPM:")));
+	bpm_enabled->connect("toggled", callable_mp(this, &AudioStreamImportSettings::_settings_changed).unbind(1));
+	interactive_hb->add_child(bpm_enabled);
+	bpm_edit = memnew(SpinBox);
+	bpm_edit->set_max(400);
+	bpm_edit->set_step(0.01);
+	bpm_edit->set_tooltip(TTR("Configure the Beats Per Measure (tempo) used for the interactive streams\n (AudioStreamTransitioner and AudioStreamPlaylist).\nIf zero, the one from the parent stream will be used."));
```

Comment:
The leading space after line break and before the parenthesis should be removed.
But I'm not sure why there's a hardcoded break, I think tooltips should be able to wrap when they reach a max width. If not, that's a bug IMO and should be changed.

Reply:
```suggestion
	bpm_edit->set_tooltip(TTR("Configure the Beats Per Measure (tempo) used for the interactive streams (AudioStreamTransitioner and AudioStreamPlaylist).\nIf zero, the value from the parent stream will be used."));
```